### PR TITLE
Adds the Monk Martial Art to the Chaplain's null rod selection

### PR DESCRIPTION
**Pull request by Iamgoofball**
:cl: Iamgoofball
add: The Chaplain has a new Martial Art available in replacement of his Null Rod. Look for the Monk Manual.
/:cl:

the Monk martial art is an advanced art not for newbies, with many fun things to decide if you hit or not, rewards for practicing, and all kinds of fun shit

it's got flurry of blows, proper fort. saves, proper armor class attack rolls, loads of shit

passive benefits include Cleave/Great Cleave and Circle Kick for that extra sexual tension on your hits

you need a lot of exp to make it to level 30

dont forget diamond body/purity of body

monks defend with proper rolls too, can dodge hits entirely

also makes disciples who cap out at level 19

### CODE REVIEW

**Comment by Iamgoofball** [comment]
>#35535

File: code/datums/martial/monk.dm [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+/datum/martial_art/monk/teach(mob/living/carbon/human/H,make_temporary=0)
+	if(..())
+		monk_rest.Grant(H)
+		to_chat(H, "<span class = 'userdanger'>You know the ancient Monk techniques!</span>")
+		to_chat(H, "<span class = 'danger'>This technique becomes more powerful the more you use it.</span>")
+		to_chat(H, "<span class = 'danger'>Your power level will increase from 1 to 20, gaining new abilities and growing stronger the more you train.</span>")
+		to_chat(H, "<span class = 'danger'>Your abilities have a use limit, before you need to rest to regain your strength.</span>")
+
+/datum/martial_art/monk/on_remove(mob/living/carbon/human/H)
+	to_chat(H, "<span class = 'userdanger'>You fprget the ways of a Monk...</span>")
```
Comment (XDTM) [inline code/datums/martial/monk.dm]: fprget

File: code/datums/martial/monk.dm [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+/datum/action/monk_rest
+	name = "Rest - Sleep to regain your strength."
+	button_icon_state = "neckchop" // todo: replace
+	icon_icon = 'icons/mob/actions/actions_items.dmi'
+
+/datum/action/monk_rest/Trigger()
+	if(owner.incapacitated())
+		to_chat(owner, "<span class='warning'>You can't use [name] while you're incapacitated.</span>")
+		return
+	owner.Sleeping(100)
```
Comment (XDTM) [inline code/datums/martial/monk.dm]: coffee will make this last a couple seconds

**Comment by Errorin** [comment]
This is what happens when you make goof play Never Winter Nights 2 and he kills everyone as monk.

File: code/datums/martial/monk.dm (hunk 1 above)
Comment (outbools) [reply code/datums/martial/monk.dm]: should be "You faget"

**Comment by Iamgoofball** [comment]
oh right I forgot about Circle Kick

**Comment by arsserpentarium** [comment]
I appreciate it more in case of chef

**Comment by Spaceinaba** [comment]
can I punch the shit out of monkies and stuff to raise my exp

**Comment by KomradeSpectre** [comment]
wtf I like goofposts now.

**Comment by davethwave** [comment]
Surprisingly fun sounding.

**Comment by Armhulen** [comment]
call it holy crunch please look you don't need to understand it but they'll ping me every day with this shit unless you do cmon goof

**Comment by KomradeSpectre** [comment]
Crunchitize me, Chaplain!

**Comment by Armhulen** [comment]
to be clear: monk is a job

holy crunch is a martial art

**Comment by Spaceinaba** [comment]
>you bald as you get stronger
hahaha

File: code/datums/martial/monk.dm [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+		to_chat(H, "<span class = 'userdanger'>You know the ancient Monk techniques!</span>")
+		to_chat(H, "<span class = 'danger'>This technique becomes more powerful the more you use it.</span>")
+		to_chat(H, "<span class = 'danger'>Your power level will increase from 1 to 20, gaining new abilities and growing stronger the more you train.</span>")
+		to_chat(H, "<span class = 'danger'>Your abilities have a use limit, before you need to rest to regain your strength.</span>")
+		START_PROCESSING(SSfastprocess, src)
+		H.hair_style = "Short Hair"
+		H.facial_hair_style = "Shaved"
+		H.update_hair()
+/datum/martial_art/monk/on_remove(mob/living/carbon/human/H)
+	to_chat(H, "<span class = 'userdanger'>You fprget the ways of a Monk...</span>")
```
Comment (GuyonBroadway) [inline code/datums/martial/monk.dm]: >"fprget"

File: code/datums/martial/monk.dm [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+	var/atr = attack_roll(D, flurry_of_blows_penalty-4)
+	if(atr)
+		var/fort_save = ROLL_DICE(1,20) // todo: figure out a good way to calc a save bonus
+		if(fort_save >= 10)
+			D.visible_message("<span class='danger'>[A] punches [D], but [D] is unfazed!</span>", \
+					  "<span class='userdanger'>[A] punches you, but you are unfazed!</span>")
+		else
+			D.Stun(30)
+			D.visible_message("<span class='danger'>[A] punches [D], stunning them!</span>", \
+					  "<span class='userdanger'>[A] misses you!</span>")
```
Comment (Armhulen) [inline code/datums/martial/monk.dm]: this is incorrect!

**Comment by Iamgoofball** [comment]
monks are basically feature complete now

**Comment by fludd12** [comment]
Can you add a way to train without attacking things, but it takes longer, so you can be one of those martial pacifist monks who don't fight except when absolutely necessary but when they do they kick ass?

Also why balding

**Comment by Wilchenx** [comment]
@fludd12 >Can you add a way to train without attacking things

he can but rewarding players for not taking risk is bad

>why balding

bald monk in robes is a iconic look

**Comment by Spaceinaba** [comment]
just go to fucking lavaland and punch the fauna to death like a REAL FUCKIN MAN

**Comment by Spaceinaba** [comment]
wait nevermind I read the code you cant go toe to toe with a goliath because PUSSY SHIT

**Comment by KomradeSpectre** [comment]
Boosting is an issue on this weapon. Players can sit and train on a person or humanized monkey (while healing them with bible) all round and become nigh unstoppable with the aoe attack in a matter of 20 minutes.

**Comment by fludd12** [comment]
I still think there should be a nonviolent training method involving, like, moving in specific ways or some weird BS, and maybe have it sync for all people with the monk martial art so you can have you and your followers resembling the generic, synchronized training thing you see in all the karate movies

**Comment by Spaceinaba** [comment]
meditating (maybe?) and making use of the weightlifting machines could give you exp (but substantially slower than combat)
idk I just feel like there should be a way to level without validhunting and getting into fights

**Comment by Iamgoofball** [comment]
I'm not adding a way for you to train outside of interacting with other people.

**Comment by fludd12** [comment]
Definitely not outside interacting with other people, but preferably without *harming* other people. Maybe some form of training session thing?

**Comment by Spaceinaba** [comment]
reasonable considering how much people hate being able to solo powergame
alternatively you could lower the level required to disciple your first disciple so you can hire a training buddy

**Comment by Iamgoofball** [comment]
I'm going to add a sparring mode toggle that stops you from hitting people who are at 25 health or below.

**Comment by Spaceinaba** [comment]
or just beat somebody up until you reach level 20 I guess then disciple them

**Comment by Spaceinaba** [comment]
oh that's a good idea

**Comment by fludd12** [comment]
That also works for me. :+1:

**Comment by factoryman942** [comment]
monks can punch through directional windows

**Comment by Spaceinaba** [comment]
feature

**Comment by Iamgoofball** [comment]
we still have directional windows?????

**Comment by Spaceinaba** [comment]
I would say "yeah dont you play" but that would be a stupid question

**Comment by fludd12** [comment]
Maybe have that be a part of diamond body, where you can immediately break through the glass pane and punch someone through it?

**Comment by Spaceinaba** [comment]
can you whitelist getting exp on lavaland monsters and megafauna so I can train in hell

**Comment by Iamgoofball** [comment]
Maybe

File: code/_onclick/item_attack.dm [hunk 5]
```diff
@@ -100,6 +105,45 @@
...
+			send_item_attack_message(I, user)
+			apply_damage(I.force, I.damtype)
+			if(I.damtype == BRUTE)
+				if(prob(33))
+					I.add_mob_blood(src)
+					var/turf/location = get_turf(src)
+					add_splatter_floor(location)
+					if(get_dist(user, src) <= 1)	//people with TK won't get smeared with blood
+						user.add_mob_blood(src)
+			return TRUE //successful attack
```
Comment (ChangelingRain) [inline code/_onclick/item_attack.dm]: this is all copypasted and makes attacked_by harder to maintain

Comment (Iamgoofball) [reply code/_onclick/item_attack.dm]: I need to kill all this because I found out this isn't even used by attack code

**Comment by Shadowflame909** [comment]
This is a good shitpost that was needed to keeping tg station fun. Thank you goof, thank you.

**Comment by YakumoChen** [comment]
Completely overpowered in the later tiers, it's a knockdown AoE with massive damage that you can basically spam for free. Goof managed to prove how OP his own PR is by running around with it as ling.

Not to mention you can't disarm it and there's no way to tell the chaplain has it at any time, and it doesn't take away use of an entire arm, so it's an obvious choice for everyone to take.

**Comment by Iamgoofball** [comment]
@YakumoChen 
I need to fix it being stackable with ling, that was hillariously absurd even for my standards.

The counter for this is literally "fire a disabler/taser" or "use a flash/stun baton" because the effects will still apply.

You also sacrifice ALL your cult immunity and cult dispelling for this.

**Comment by Iamgoofball** [comment]
for reference, I was running around as a changeling with adrenals, a good stockpile of chems, and fleshmennd, and abusing t he fact the crew didn't know I was a changeling to get the drop on them with revives. My early kills also abused the mute sting.

**Comment by Iamgoofball** [comment]
also you can disarm it; remove the arm

**Comment by YakumoChen** [comment]
"Disarm the arm" is basically saying "just kill them lmao that will end it"

Having an invisible, incredibly powerful weapon (like carp) is not something crew should get every shift, for free.

**Comment by YakumoChen** [comment]
You also get a free, invisible stun that recharges, for free.

**Comment by Iamgoofball** [comment]
A stun baton that misses hilariously often and is almost a guaranteed miss against anyone with armor.

**Comment by YakumoChen** [comment]
And a free, invisible, free heal.

The more I look at this, the more it becomes basically ling junior

**Comment by Iamgoofball** [comment]
A heal that requires you to become incredibly vulnerable to enable another use of. Unlike fleshmend.

I might remove wholeness though it was pretty crap

**Comment by Iamgoofball** [comment]
Wholeness = the heal

**Comment by davethwave** [comment]
Honestly I enjoy this addition

**Comment by fludd12** [comment]
I'd be fine with making the heal a channel, but please don't remove it!

**Comment by Robustin** [comment]
Quivering palm refreshing upon sleep is silly though.

**Comment by coiax** [comment]
This is really quite silly. I appreciate the reference to D&D 3.5 edition Monks, but you can kill someone in approximately 3 hits.

Not to mention how all latejoin Chaplains can pick this, and then you've just got a horde of monks running around beating each other up.

**Comment by Iamgoofball** [comment]
>all latejoin Chaplains can pick this

There's only 1 null rod dude

**Comment by coiax** [comment]
that's not true.

**Comment by Spaceinaba** [comment]
no there's only one null rod do you even play coiax?

**Comment by davethwave** [comment]
Unless latejoin chaplains start with a null rod there is only one on station

**Comment by Spaceinaba** [comment]
they dont

**Comment by Robustin** [comment]
ITT nobody plays this fucking game.

Latejoin chaplains get whatever rod the original chose. We had 10 kungfu chaplains earlier.

**Comment by Iamgoofball** [comment]
Jesus christ

I'm going to remove that, dear god

**Comment by PKPenguin321** [comment]
It probably shouldn't be removed considering it was intentional but maybe they should all just get to choose what null rod they want instead of being stuck with what the original chose

**Comment by PKPenguin321** [comment]
what

**Comment by Iamgoofball** [comment]
also add Cook to that list until my or naksu's nerf PR gets merged

**Comment by PKPenguin321** [comment]
ok?

**Comment by Iamgoofball** [comment]
are you not seeing the problem with "on paper and in practice, why would I pick any job other than chaplain/cook/security officer for maximum powergame"

if they can powergame it, they will, this is a rule of SS13 design

**Comment by Iamgoofball** [comment]
@Spaceinaba It either uses your backup human name you set in character creation or randomizes a new name for you.

**Comment by Spaceinaba** [comment]
ah okay

**Comment by PKPenguin321** [comment]
ok but goof it's already in the game deliberately so removing it would kind of leave the scope of this PR (and i dont really ever see it getting abused)
however, making it so chaplains are allowed to do their own thing if they latejoin instead of being forced into doing wahtever the first chaplain did could totally be in this PR

**Comment by YakumoChen** [comment]
>why wouldn't I just powergame and close all slots that aren't security and chaplain?

Isn't this just self-admitting that you're making the chaplain overpowered, goof

**Comment by Iamgoofball** [comment]
re-read what I said my dude, chaplains are already the meta pick

**Comment by AnturK** [comment]
Dunno if fixed already didn't read the code yet, `Runtime in monk.dm, line 50: type mismatch: "Cat" -= "Cat"`

**Comment by ThePainkiller** [comment]
Found a bug:

**Round ID:** #83597 (Sybil, Omegastation)

Dying and getting cloned does not reset your "ki levels" (punches stay the same damage, penetrate armor, cleave attack, etc), but your HUD buttons from stun punch onwards completely disappear.

Also, your name gets randomly changed EVERY TIME you clone, and you cannot equip headgears, not even hardsuit ones (is this last one intended?)

**[Reproduction]:** The exact thing is that I died to a normal Legion near a tendril and got absorbed by it. A lizard got me out, cloned me on a stock cloner (i.e., not upgraded at all) back in the station, and when I finished cloning, I had lost all my buttons (the stun punch, chakra and heart stopping punch) including the one to make a disciple (I had already made my first disciple at this point)

**Comment by davethwave** [comment]
I was thinking of that instead of healing with only needing rest but that when you do you gain a few debuffs from doing so like making you hungry or by relying on it too much it would give the monk permanent stamina damage instead of normal medicine if your low leveled

**Comment by AdamElTablawy** [comment]
kung fu lings really need to die tbh, i got it and it was ridiculous

**Comment by Iamgoofball** [comment]
working on a fix for that tonight

File: code/__DEFINES/maths.dm [hunk 6]
```diff
@@ -91,6 +91,9 @@
 // Returns the nth root of x.
 #define ROOT(n, x) ((x) ** (1 / (n)))
 
+// Real modulus that handles decimals
+#define ROLL_DICE(amt, faces) (rand(0, (faces-1)*amt) + amt)
```
Comment (FlattestGuitar) [inline code/__DEFINES/maths.dm]: This is not how you roll dice. The probability is evenly distributed and it shouldn't be.

Easiest fix I can think of would be to outsource this to some nice old lady in China for $1 a day.

**Comment by improvedname** [comment]
Daily reminder that everyone can grab a chair and fuck a monk up

**Comment by GuyonBroadway** [comment]
Chair-fu is a SORELY underused fighting style

File: code/__DEFINES/maths.dm (hunk 6 above)
Comment (Iamgoofball) [reply code/__DEFINES/maths.dm]: We dont use this anymore

**Comment by TimeIsEndless** [comment]
Currently the Monk Manual turns you human upon use. This in its own isnt too bad but it also does not grant you the ability to rest severly hampering the use of the Monk Powers. Also most Hud buttons seem to disappear at random

**Comment by Slignerd** [comment]
Nothing beats joining into the round only to be one hit critted by the chaplain this round.

Seriously, please don't.  It's like sleeping carp memes, but now it happens every single round, and you don't even have to be an antag to get it.

**Comment by kevinz000** [comment]
right :-1: way too strong for a nonantag item and apparently you can make disciples so the chaplain isn't theo nly guy with these powers so way way too strong.

**Comment by Iamgoofball** [comment]
@slignerd Have you considered ahelping them for camping the shuttle?

If that wasn't the case, try applying literally any ranged weapon or stun in the game

@kevinz000 Have you tried applying any ranged weapon or stun in the game?

**Comment by Supermichael777** [comment]
Don't force-human non-command jobs just because you hate cat/lizard/moff

**Comment by Iamgoofball** [comment]
@Supermichael777 we've been over this, please read the thread, it's to force bald

**Comment by Iamgoofball** [comment]
I need to blacklist those.

**Comment by Mey-Ha-Zah** [comment]
@Iamgoofball Problem with forcing bald, is how do people use space suits in an emergency?

Or the fact that chap cloak, hardsuit and other hooded things completely ignore the no headwear thing.

Maybe just make wearing headgear disable monk abilities. That way people can be whatever species they want.

**Comment by Iamgoofball** [comment]
>is how do people use space suits in an emergency?

This is one of the things you trade for your powers.

>Or the fact that chap cloak, hardsuit and other hooded things completely ignore the no headwear thing.

This is a bug.

>Maybe just make wearing headgear disable monk abilities. That way people can be whatever species they want.

Nope.

**Comment by YakumoChen** [comment]
Readd guns to svi if it's too powerful you can just kill the guy with guns right?

**Comment by Shadowflame909** [comment]
If you see the chaplain murdering a guy with kung fu as sec. Just taser that nerd and cuff him, he literally cannot break out of anywhere with his kung fu alone. It only works on people.

**Comment by YakumoChen** [comment]
Also force all scientists that print guns to be catgirls, that's just a trade-off you have to live with if you wanna hav OP stuff.

**Comment by Shadowflame909** [comment]
Force all security officers to wear grey jumpsuits if they harmbaton. Trade off for being a nerd.

**Comment by YakumoChen** [comment]
Are the guns invisible too. Don't worry, catgirls are a meta tell they have guns

**Comment by Spaceinaba** [comment]
allow nonhuman chaplains before I destroy you irl

**Comment by Kraseo** [comment]
I think forcing humanize on a mob just to make their hairstyle bald is the stupidest excuse I've ever heard this year yet

**Comment by YakumoChen** [comment]
Level 20 kek fu: all shots home in on other players

**Comment by kevinz000** [comment]
what kraso said

**Comment by YakumoChen** [comment]
Objectively botany is a meta pick you should be every round, they can grow exploding fruit

**Comment by Wilchenx** [comment]
Yakumochen meltdown

**Comment by Slignerd** [comment]
@Iamgoofball There wasn't any shuttle camping, it's just that it's bullshit.

"try applying a ranged weapon" doesn't work if you don't know that the guy is dangerous to begin with - and by the time he punches you and you realize that, it's too late.

**Comment by Iamgoofball** [comment]
i'll consider making the student robes a requirement alongside the baldness

**Comment by Spaceinaba** [comment]
allow nonhumans

**Comment by Slignerd** [comment]
Baldness doesn't even work. All I had to do was use a mirror, and I got my hair back right away.

**Comment by Iamgoofball** [comment]
mirror exploit will be patched, please read the thread you are commenting in

**Comment by YakumoChen** [comment]
On a serious note the leveling mechanic just encourages you to park at assistant near a medibot and honor town until you get all the superpowers

**Comment by Iamgoofball** [comment]
so what you're saying is I added inter-departmental cooperation?

**Comment by Spaceinaba** [comment]
allow nonhumans goof I need to get my rocks off by fisting people as a lizard

**Comment by Iamgoofball** [comment]
@Spaceinaba please stop repeating the same thing or I'll ask oranges to delete duplicate comments, this isnt the hut

**Comment by Kraseo** [comment]
damn goofball your frontal lobe must be fucking swole from all those mental gymnastics you're doing

**Comment by Iamgoofball** [comment]
i know, right?

**Comment by Spaceinaba** [comment]
they're not duplicate comments I keep adding more words to them

**Comment by YakumoChen** [comment]
If you need all this snowflake code tore players to be bald humans maybe the problem is forcing them to be bald humans

**Comment by Wilchenx** [comment]
If you think there's something wrong with being bald tell that to MSO

**Comment by Iamgoofball** [comment]
nice attempt, but it's not snowflake my dude, this is a pretty basic species change

**Comment by YakumoChen** [comment]
If you need to add exceptions to mirrors to prevent not going bald it's snowflake my dude

**Comment by Iamgoofball** [comment]
nah m8, not snowflake, i'm gonna add a species flag for it

**Comment by YakumoChen** [comment]
So you made a snowflake species for monks?

**Comment by Iamgoofball** [comment]
are you OK? you seem to be reaching much further than you usually do

**Comment by YakumoChen** [comment]
Sounds like I'm right then

**Comment by Iamgoofball** [comment]
no, actually, you're incredibly far off

**Comment by YakumoChen** [comment]
It's still snowflake checks and code just to accommodate one martial art that's basically free carp

**Comment by Iamgoofball** [comment]
changing your argument every time is just kinda sad dude

**Comment by Spaceinaba** [comment]
why are you racist anyways goof are you a closet furry

**Comment by ThePainkiller** [comment]
So, this got de-merged because people whined too much about something that they don't know how to play against? They consider a martial art that makes you unable to use hardsuit headgears, any item as a melee weapon, and ranged weaponry OP just because it can crit you in three punches wich are easily blockable by holding a literally fucking chair in your hand? It doesn't even reflect bullets and thrown weapons/items like carp for fucks sakes, it's just a buffed punches CQC and even its special effects were all chance-based.
Let me put it on meme format:
_"Hahahahahahahaha How The Fuck Is Monk OP Hahahaha Ligga Just Grab A Chair Like Ligga He Can Still Be Stunned And Killed With Ranged Weaponry Haha"_

**Comment by YakumoChen** [comment]
Consider: a chair is not likely to help when the bald dude can run up to you with no weapons in hand and punch you to death before you have a chance to even realize what's up.

What do you think makes carp so strong?

**Comment by ThePainkiller** [comment]
If you see a Chaplain with his weapon either not embedded in his own hand nor around his waist/back then assume he's either rogue as fuck or a monk user, or both. You can hold a bar/kitchen stool for the same effect and you don't even need any special access to a certain area, nor do you have any deterrents to get one. You even get a message if he prepares an attack stance like stun punch and then again it was **_chance based_**
_"What do you think makes carp so strong?"_
Gee I dunno, maybe the fact that being immune to all fucking kinds of ranged attacks forces you to face them on melee, in wich they're stronger unless you have a stunbaton/telescopic baton, or using flashbangs unless they have protection against them?
And Carp's greatest damaging move / instakill requires the target to be downed and four uninterrumpted hits changing intents.
But it's easier to blame one's own unrobustness on a game mechanic and call it _"broken"_.

What about Xenos who have a 100% stun chance on disarm, an ebow spit, insane speed, chameleon capabilities, wall melting capabilities, are space proof, do not breathe, integrated jaws of life, immune to diseases, and can flat out win a fight by pregging you with a facehugger? Fair and balanced, right?

**Comment by YakumoChen** [comment]
Xenos are supposed to be round-ending. Xenos are antags and not comparable to a guy who can use a roundstart exclusive item to become comparable to an item that costs a traitor all their TC.

**Comment by optimumtact** [comment]
another goof meme he runs into the ground

**Pull request closed without merging**